fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -178,7 +181,8 @@ public OptExpression rewrite() {
 
         ScalarOperator mvPredicate = MvUtils.rewriteOptExprCompoundPredicate(mvExpression, mvColumnRefRewriter);
 
-        if (materializationContext.getMvPartialPartitionPredicate() != null) {
+        if (materializationContext.getMvPartialPartitionPredicate() != null
+                && !(mvPredicate instanceof IsNullPredicateOperator)) {
```

Comment:
If non partition column has predicate `isNull`, it will do not add latest partition predicate to mv predicate?

Reply:
if there is IsNullPredicateOperator for partition column, should not add mv partial partition predicate.